src/film.js

```diff
@@ -19,7 +21,64 @@ export default class Film extends Component {
 
     this._element = null;
     this._onComments = null;
+    this._getId = null;
+
+    this._onAddToWatchList = null;
+    this._onAddToFavorite = null;
+    this._onMarkAsWatched = null;
+
     this._onCommentsButtonClick = this._onCommentsButtonClick.bind(this);
+    this._onWatchListButtonClick = this._onWatchListButtonClick.bind(this);
+    this._onFavoriteButtonClick = this._onFavoriteButtonClick.bind(this);
+    this._onWatchedButtonClick = this._onWatchedButtonClick.bind(this);
+    this._onAnyBtnClick = this._onAnyBtnClick.bind(this);
+  }
+
+  set getId(fn) {
+    this._getId = fn;
+  }
+
+  _onAnyBtnClick() {
+    return typeof this._getId === `function` && this._getId(this._id);
+  }
+
+  set onAddToFavorite(fn) {
+    this._onAddToFavorite = fn;
+  }
+
+  _onFavoriteButtonClick(evt) {
+    evt.preventDefault();
+    this._isFavorite = !this._isFavorite;
+    const FavoriteState = this._isFavorite;
+    if (typeof this._onAddToFavorite === `function`) {
+      this._onAddToFavorite(FavoriteState);
+    }
+  }
+
+  set onAddToWatchList(fn) {
+    this._onAddToWatchList = fn;
+  }
+
+  _onWatchListButtonClick(evt) {
+    evt.preventDefault();
+    this._isWatchList = !this._isWatchList;
+    const WatchListState = this._isWatchList;
```

Comment:
Критерий [Б3](https://up.htmlacademy.ru/ecmascript/8/criteries#b3):
```suggestion
    const watchListState = this._isWatchList;
```